src/autogluon/cloud/job/sagemaker_job.py

```diff
@@ -265,6 +267,10 @@ def run(
         else:
             model_cls = AutoGluonNonRepackInferenceModel
         logger.log(20, "Creating inference model...")
+        inference_kwargs_str = json.dumps(inference_kwargs) if inference_kwargs is not None else None
```

Comment:
Can we support complex payload in the predict args such as `known_covariates` from [TimeSeriesPredictor](https://auto.gluon.ai/stable/api/autogluon.timeseries.TimeSeriesPredictor.predict.html)

Reply:
We might need to handle DataFrame specifically. I can add that to the PR.